### PR TITLE
LPD-26298 Custom date filter in Content Dashboard

### PR DESCRIPTION
**Pull request by jkappler**
Custom date filter in Content Dashboard

# What is this trying to solve?

[Link to ticket](https://issues.liferay.com/browse/LPD-26298)

The PR adds the option to sort the results in the content dashboard by specific date ranges 

# How am I fixing it?

Modify the logic that we currently have in content dashboard to be able to filter by a range of dates of a given date type

# How can you verify that it works?

The PR includes a set of unit tests

1. An unit test to check that when the correct parameters are set, they are taken into account when building the search context
2. An unit test to check that when a non existing date type is selected, no filters are applied
3. An unit test to check that when non valid dates are selected, no filters are applied

The PR can also be tested through the following steps

1. Add a new Web Content or image
2. Go to Content Dashboard
3. In the toolbar, select Custom Date 
4. In the popup, select "Create Date"
5. Select the range of the dates
6. Accept
7. If the range of dates includes the web content, it will be shown

This is nice to test with a portal that has content created throughout several dates, to see the real effect from the filter. The database can also be manipulated manually to change the dates

filter

@sandrodw3

### CODE REVIEW

**Comment by jkappler** [comment]
ci:test:sf

**Review by Mikellorza — APPROVED** [review]
LGTM!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		const url = new URL(filterUrl);
+
+		url.searchParams.set(`${namespace}dateType`, dateType!);
+		url.searchParams.set(`${namespace}startDate`, startDate!);
+		url.searchParams.set(`${namespace}endDate`, endDate!);
+
+		navigate(url.toString());
+	};
+
+	if (!open) {
```
Comment (ambrinchaudhary) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx:84]: Is this needed? I tested with this removed and it works fine.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+								setStartDate(start);
+								setEndDate(end);
+
+								setHasError(false);
+							}}
+							placeholder="YYYY-MM-DD - YYYY-MM-DD"
+							range
+							value={getDateRange(startDate, endDate)}
+							years={{
+								end: 2024,
```
Comment (ambrinchaudhary) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx:129]: I guess that this shouldn't be hardcoded. `end: new Date().getFullYear()` would be the best, and maybe `new Date().getFullYear() - 10` ? for the start date?

**Review by ambrinchaudhary — CHANGES_REQUESTED** [review]
Just some minor comments. Otherwise it LGTM.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx (hunk 2 above)
Comment (sandrodw3) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx:129]: Totally agree, copied it from the Clay example and forgot to change it 😅 will adapt it, thanks Ambrin!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx (hunk 1 above)
Comment (sandrodw3) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx:84]: Um, probably no, will make sure of it and remove it 🥰

**Comment by sandrodw3** [comment]
@ambrinchaudhary changes applied, thanks a lot for your review 🥰

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward:force

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**